packages/cdk/array/src/by-property-value.ts

```diff
@@ -0,0 +1,6 @@
+export function byPropertyValue<ArrayEntryType, ArrayEntryTypeKey extends keyof ArrayEntryType>(
```

Comment:
prefer for DX
`byPropertyValue('a', 'title')` -> `findByWhere({ key: 'a', value: 'title' })`

Reply:
you are sure?

```ts
[{ id: 1, draft: true }, { id: 2, draft: false }].find(byPropertyValue('draft', true)) 
[{ id: 1, draft: true }, { id: 2, draft: false }].filter(byPropertyValue('draft', true)) 
```
vs
```ts
[{ id: 1, draft: true }, { id: 2, draft: false }].find(findByWhere({ key: 'draft', value: true })) 
[{ id: 1, draft: true }, { id: 2, draft: false }].filter(findByWhere({ key: 'draft', value: true })) 
```

Reply:
funny semblance of jest:
```ts
const notes = [{ id: 1, draft: true }, { id: 2, draft: false }];
notes.find(whereProperty('draft').is(true))
notes.find(whereProperty('draft').isTrue()) 
notes.find(whereProperty('id').isNot(2)) 
notes.find(whereProperty('id').isIn([1, 2])) 
notes.find(whereProperty('id').isNotIn([1, 2]))

Reply:
```ts
notes.find(where(property('id').isBetween(1, 2) |> and |> property('draft').isTrue()))) 
```
:peka:

Reply:
sounds good with your examples

Reply:
maybe, can we use a jest utils style for filter behavior?

Reply:
i mean, we won, but the solution is too complex

Reply:
Okey, not problem

also next PR you can add `whereProperty` from your playground for as an alternative function `byPropertyValue`